test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -1,181 +1,84 @@
-        self.assertEqual(self.merge_dict["a"], {100, 1001})
-        self.assertEqual(len(self.merge_dict), 1)
-        # NOTE: remove_custom removes the whole overloads
-        self.merge_dict.remove_custom("a")
-        self.assertEqual(len(self.merge_dict), 0)
-        self.assertNotIn("a", self.merge_dict)
-
-    def test_bool_is_true_when_not_empty(self):
-        if self.merge_dict:
-            self.fail("MergeDict should be false when empty")
-        self.merge_dict.add_custom("a", 1)
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-        self.merge_dict.set_base("a", 42)
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-        self.merge_dict.remove_custom("a")
-        if not self.merge_dict:
-            self.fail("MergeDict should be true when not empty")
-
-
-class TestRegistrationDecorators(common_utils.TestCase):
+
+class TestRegistration(common_utils.TestCase):
     def setUp(self) -> None:
         self.registry = registration.OnnxRegistry()
+        self.custom_domain = onnxscript.values.Opset(domain="custom", version=1)
 
     def tearDown(self) -> None:
         self.registry._registry.pop("test::test_op", None)
 
-    def test_onnx_symbolic_registers_function(self):
-        self.assertFalse(self.registry.is_registered_op("test::test_op", 9))
+    def test_onnx_custom_symbolic_registers_function(self):
+        self.assertFalse(self.registry.is_registered_op("test", "test_op", "default"))
 
-        def test(g, x):
-            return g.op("test", x)
-
-        self.registry.register("test::test_op", 9, test, custom=True)
-        self.assertTrue(self.registry.is_registered_op("test::test_op", 9))
-        function_group = self.registry.get_function_group("test::test_op")
-        assert function_group is not None
-        self.assertEqual(function_group.get(9), {test})
+        @onnxscript.script(self.custom_domain)
+        def test(x, y):
+            return op.Add(x, y)
 
-    def test_custom_onnx_symbolic_registers_custom_function(self):
-        self.assertFalse(self.registry.is_registered_op("test::test_op", 9))
+        self.registry.register_custom_op(test, "test", "test_op", "default")
+        self.assertTrue(self.registry.is_registered_op("test", "test_op", "default"))
 
-        def test(g, x):
-            return g.op("test", x)
-
-        self.registry.register("test::test_op", 9, test, custom=True)
+        function_group = self.registry.get_functions("test", "test_op", "default")
+        assert function_group is not None
+        self.assertEqual({func.onnx_function for func in function_group}, {test})
 
-        self.assertTrue(self.registry.is_registered_op("test::test_op", 9))
-        function_group = self.registry.get_function_group("test::test_op")
+        function_group = self.registry._get_custom_functions(
+            "test", "test_op", "default"
+        )
```

Comment:
Duplicate of L43-45?

Reply:
Tried on two different functions

Reply:
I see. Not very obvious the first sight so I would see if there’s a way to make it more apparent

Reply:
_get_custom_functions is deleted, as we have get_functions return order list.